### PR TITLE
[tests] have most tests use pytest test runner

### PR DESCRIPTION
**Pull request by brettlangdon**
This PR updates most test to use `pytest` instead of `nosetests`.

I had some issues with some tests running with `s/nosetests/pytest/`, so I did not move all of them.

Changes:
- `s/nosetests/pytest/` on `tox.ini`
- Move `tests/base.py` -> `tests/base/__init__.py`
- Add `conftest.py` to determine when to run `py35` tests or not

The way we were excluding tests was ignoring test cases that also matched the patterns, but that doesn't happen with `pytest` anymore.

Worth noting that `pytest` can't ignore specific files, only directories... 🙄

### CODE REVIEW

**Comment by brettlangdon** [comment]
I need to add a comment to `conftest.py` on what it is.

File: tests/test_global_config.py [hunk 1]
```diff
@@ -40,14 +40,14 @@ def test_settings_copy(self):
         ok_(self.config.requests['distributed_tracing'] is True)
         ok_(self.config.requests['experimental']['request_enqueuing'] is True)
 
-    def test_missing_integration(self):
+    def test_missing_integration_key(self):
```
Comment (majorgreys) [inline tests/test_global_config.py]: Why this change here in this PR? I'm not sure whether we just are no longer raising `ConfigException`.

Comment (brettlangdon) [reply tests/test_global_config.py]: so... this is fun... under `nosetests` this test was never getting executed. We were ignoring any tests that matches the regex we were supplying, directories, test case classes, and test functions. (we were trying to ignore the "integration" directory).

And since the test was never running we never noticed that it wasn't testing the correct thing. It is testing that `config.unknown` will fail, but it doesn't fail anymore, we automatically add an entry for any integration we attempt to access. So instead I made it where if we try to access an integration setting that does not exist.

Comment (Kyle-Verhoog) [reply tests/test_global_config.py]: Hopefully when we introduce some kind of coverage tool we'll be able to detect when our tests aren't running to avoid something like this in the future.

**Comment by brettlangdon** [comment]
@Kyle-Verhoog I rewrote `conftest.py`, better comments, but also went for an approach that wouldn't require us to manually list directories/versions for ignoring

File: conftest.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+PY_DIR_PATTERN = re.compile(r'^py[23][0-9]$')
+
+
+# Determine if the folder should be ignored
+# https://docs.pytest.org/en/3.10.1/reference.html#_pytest.hookspec.pytest_ignore_collect
+# DEV: We can only ignore folders/modules, we cannot ignore individual files
+# TODO: Should this be `pytest_collect_directory`? or `pytest_collect_file`?
+# DEV: We must wrap with `@pytest.mark.hookwrapper` to inherit from default (e.g. honor `--ignore`)
+#      https://github.com/pytest-dev/pytest/issues/846#issuecomment-122129189
```
Comment (Kyle-Verhoog) [inline conftest.py]: :-1: annoying but I guess we gotta live with it.

File: conftest.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    Example::
+
+        File: tests/contrib/vertica/py35/test.py
+        Python 2.7: Skip
+        Python 3.4: Skip
+        Python 3.5: Collect
+        Python 3.6: Collect
+
+    :rtype: bool
+    :returns: ``True`` to skip the directory, ``False`` to collect
```
Comment (Kyle-Verhoog) [inline conftest.py]: Unless I'm missing something, I don't see a `True` or `False` being returned by this function. Does it work?

Comment (brettlangdon) [reply conftest.py]: Whoops, good catch. Since wrapping it doesn't need to return anymore, just update `outcome.force_result()`

File: conftest.py (hunk 2 above)
Comment (brettlangdon) [reply conftest.py]: yeah, either that or code support for `--ignore` ourselves 🙄

File: conftest.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    Example::
+
+        File: tests/contrib/vertica/py35/test.py
+        Python 2.7: Skip
+        Python 3.4: Skip
+        Python 3.5: Collect
+        Python 3.6: Collect
+    """
+    # Execute original behavior first
+    # DEV: We need to set `outcome.force_result(True)` if we need to override
```
Comment (Kyle-Verhoog) [inline conftest.py]: :+1:

File: conftest.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import pytest
+
+PY_DIR_PATTERN = re.compile(r'^py[23][0-9]$')
+
+
+# Determine if the folder should be ignored
+# https://docs.pytest.org/en/3.10.1/reference.html#_pytest.hookspec.pytest_ignore_collect
+# DEV: We can only ignore folders/modules, we cannot ignore individual files
+# TODO: Should this be `pytest_collect_directory`? or `pytest_collect_file`?
```
Comment (Kyle-Verhoog) [inline conftest.py]: Do we need this TODO? I think maybe directory makes more sense since we can't ignore individual files but not a big deal imo.

Comment (brettlangdon) [reply conftest.py]: We *might* be able to ignore individual files, just not sure if we want to or not.

Having as their own `py35/` directory means we can also selectively run `flake8` for Python3 files as well.
